doc/releases/changelog-dev.md

```diff
@@ -152,6 +152,12 @@
 
 **Operator Arithmetic:**
 
+* `default.qubit` now will natively execute any operation that defines a matrix except
+  for trainable `Pow` operations. This includes custom operations, `GroverOperator`, `QFT`,
+  `U1`, `U2`, `U3`, and arithmetic operations. The existance of a matrix is determined by the
```

Comment:
Should we benchmark just to ensure that there are no edge cases where this could impact certain workflows?

Reply:
For local experimentation, natively supporting the `U` gates provides a large performance increase.

For less than 15 qubits, `GroverOperator` is faster using the matrix than the decomposition. `QFT` was already natively supported.

Reply:
Thanks Christina! Two things come to mind:

- this might affect `lightning.qubit`, since it currently inherits from `default.qubit`. Not ideal, and something we should address [@mlxd]

- As a result, it would probably be good to test this in the >20 qubits regime, where I expect this to make a difference. Something that was flagged by users is that while computing Hamiltonians using sparse techniques is much faster for <20 qubits, it becomes an huge bottleneck >20 qubits.

Reply:
![image](https://user-images.githubusercontent.com/858615/181833133-f620f854-8896-4df2-8ec1-c4fe6e524671.png)

Results from Lightning show no appreciable change in performance in either direction, all the way up to 26 qubits with adjoint.

Circuit as listed below:
```python
import numpy as np
import pennylane as qml
import random
import sys
import time
from pennylane.templates import BasicEntanglerLayers
from pennylane import numpy as pnp
from numpy.random import random
np.random.seed(42)

if len(sys.argv) != 3:
    print("python benchmark.py <num_qubits> <num_runs>")
    sys.exit(1)

n_wires = int(sys.argv[1])
runs = int(sys.argv[2])
device_name = "lightning.qubit"
diff_method = "adjoint" 
dev = qml.device(device_name, wires=n_wires)

@qml.qnode(dev, diff_method=diff_method)
def circuit(weights):
    for i in range(n_wires):
        qml.RX(weights[0], wires=i)
        qml.RY(weights[1], wires=i)
        qml.RZ(weights[2], wires=i)
        qml.IsingXX(weights[3], wires=[i,(i+1)%n_wires])
        qml.IsingYY(weights[4], wires=[i,(i+1)%n_wires])
        qml.IsingZZ(weights[5], wires=[i,(i+1)%n_wires])
        qml.SingleExcitation(weights[6], wires=[i,(i+1)%n_wires])
        qml.DoubleExcitation(weights[7], wires=[i,(i+1)%n_wires,(i+2)%n_wires, (i+3)%n_wires])
    return [qml.expval(qml.PauliZ(i)) for i in range(n_wires)]

init_weights = np.random.random(8) 
params = pnp.array(init_weights,requires_grad=True)
results = []
times = []
for vals in range(runs):
    start = time.time()
    jac = qml.jacobian(circuit)
    results.append(jac(params))
    end = time.time()
    times.append(end-start)
print(n_wires, np.mean(times), np.std(times), np.mean(results))

```